common/common.go

```diff
@@ -679,3 +679,39 @@ func (c *Counter) IncrementAndGet() int64 {
 	}
 	return newID
 }
+
+// ElementProcessor defines the function signature for processing an individual element with its index.
+// Index required in old subscription matching // TODO: Remove this when gateio templates are updated to all assets.
+type ElementProcessor[E any] func(index int, element E) error
+
+// ThrottledBatch takes a slice of elements and processes them in batches of `batchSize` concurrently.
+// For example, if batchSize = 10 and list has 100 elements, 10 goroutines will process 10 elements concurrently
+// in each batch. Each batch completes before the next batch begins.
+// `process` is a function called for each individual element with its index and value.
+func ThrottledBatch[S ~[]E, E any](batchSize int, list S, process ElementProcessor[E]) error {
```

Comment:
No throttling occurs here. You are not analysing and/or limiting throughput based on anything. I don't think ThrottledBatch works

Reply:
Yep. I think I'm still [here](https://github.com/thrasher-corp/gocryptotrader/pull/1722#discussion_r1893400260) but I need to parse this again.
I'll clear my stack and do that on Sunday.